streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableJoin.java

```diff
@@ -42,6 +44,19 @@
                 return new KTableKTableJoinValueGetter(valueGetterSupplier1.get(), valueGetterSupplier2.get());
             }
 
+            @Override
+            public String[] storeNames() {
```

Comment:
Any reason why we don't just make this method return `List<String>` or `Collection<String>` ?